specification/glossary.md

```diff
@@ -0,0 +1,50 @@
+# Glossary
+
+Below are a list of some OpenTelemetry specific terms that are used across this
+specification.
+
+## (Telemetry) Library
+
+Denotes the actual *OpenTelemetry* (or API compatible) library in use.  
+See [Library](data-resource-semantic-conventions.md#library)
+
+<a name="instrumented_library"></a>
+
+## Instrumented Library
+
+Denotes the library whose "actions" (e.g. called methods/functions, operations, ...)
+result in OpenTelemetry signals (traces, metrics, logs).  
+The actual OpenTelemetry API calls can happen inside this library but might as well
+happen in a different one (the [Instrumenting Library](#instrumenting_library)).
+
+Example: `org.mongodb.client`.
+
+<a name="instrumenting_library"></a>
+
+## Instrumenting Library
+
+Denotes the library which provides the instrumentation *for* a given library (i.e.
+the [Instrumented Library](#instrumented_library)). Hence, *Instrumented Library*
```

Comment:
I would leave out the part in parantheses, it is not 100% clear that it applies to the "given library" and not the "library which provides".

Reply:
Using *for* again should clear that up.